### ISSUE TITLE
validate_syntax and start_domain_test functions do not correctly validate IPv4 and IPv6 addresses

### ISSUE THREAD

**Issue by matsduf** [dotse/zonemaster-backend#173]
Both validate_syntax and start_domain_test validate the parameters, e.g. the IPv4 address. The validation accepts truncated IPv4 address, i.e. addresses of 1-3 octetts. When a test started with a truncated IPv4 address is fetched, the following is returned:
{"id":"2016:000006","error":{"message":"malformed JSON string, neither array, object, number, string or atom, at character offset 0 (before \"(end of string)\") at /usr/local/share/perl/5.18.2/Zonemaster/WebBackend/DB/MySQL.pm line 200.\n","code":-32603},"jsonrpc":"2.0"}

Validation is done in WebBackend/Engine.pm with Net::IP::XS::ip_is_ipv4 which accepts truncated IPv4 addresses, which is documented in the man page.

We have the same problem with IPv6 addresses where the last two IPv6 address "words" are written in IPv4 format, e.g. "2a00:801::255.255.255.254". If the IPv4 part is complete everything works as it should, but if the IPv4 part is truncated, e.g. "2a00:801::255.255.255", it is accepted by ip_is_ipv6() (since it uses ip_is_ipv4() to parse the last part), but it will fail in zonemaster.

The validation should be stricter.

**Comment by sandoche2k** [comment]
@matsduf to provide the Regexp. 
@mtoma to implement

**Comment by matsduf** [comment]
@mtoma, if the following regexp is compined with test of Net::IP then only valid IPs will pass:

```
/^(([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})|([0-9A-Fa-f]{1,4}:[0-9A-Fa-f:]{1,}(:[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})?)|([0-9A-Fa-f]{1,4}::[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))$/
```

**Comment by sandoche2k** [comment]
@mtoma thanks for taking a look at this issue.

**Comment by sandoche2k** [comment]
closed by #328
